js/src/visitor/vectorloader.ts

```diff
@@ -178,6 +179,13 @@ export class JSONVectorLoader extends VectorLoader {
             return packBools(sources[offset] as number[]);
         } else if (DataType.isUtf8(type) || DataType.isLargeUtf8(type)) {
             return encodeUtf8((sources[offset] as string[]).join(''));
+        } else if (DataType.isInterval(type)) {
+            if (type.unit === IntervalUnit.MONTH_DAY_NANO) {
+                return encodeIntervalMonthDayNano(sources[offset]);
+            }
+            if (type.unit === IntervalUnit.DAY_TIME) {
+                return encodeIntervalDayTime(sources[offset]);
+            }
```

Comment:
This seems... wrong. The Arrow IPC reader shouldn't need to decode any values. These should be accessible via zero-copy subarray slices of the data buffer.

Reply:
Nevermind, didn't realize this was in the `JSONVectorLoader`.